code/game/objects/items/weapons/towel.dm

```diff
@@ -10,12 +10,24 @@
 	hitsound = 'sound/weapons/towelwhip.ogg'
 
 /obj/item/weapon/towel/attack_self(mob/living/user as mob)
-	user.visible_message("<span class='notice'>\The [user] uses \the [src] to towel themselves off.</span>")
-	playsound(user, 'sound/weapons/towelwipe.ogg', 25, 1)
-	if(user.fire_stacks > 0)
-		user.fire_stacks = (max(0, user.fire_stacks - 1.5))
-	else if(user.fire_stacks < 0)
-		user.fire_stacks = (min(0, user.fire_stacks + 1.5))
+	attack(user,user)
+
+/obj/item/weapon/towel/attack(mob/living/carbon/human/M as mob, mob/living/carbon/user as mob)
+	if(istype(M) && user.a_intent == I_HELP)
+		user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+		if(user.on_fire)
+			user.visible_message("<span class='warning'>\The [user] uses \the [src] to pat out \the [M]'s flames with \the [src]!</span>")
```

Comment:
See below. I don't thing `\the` should be used for mobs with proper noun names. This might be automatically handles by BYOND, though, so who knows.